src/components/Popup/Popup.tsx

```diff
@@ -63,6 +71,9 @@ export interface PopupProps
    */
   offset?: string
 
+  /** Events triggering the popup. */
+  on?: PopupEvents | PopupEventsArray
```

Comment:
This type allows this:

```ts
on={['click', 'click']}
```

[TS playground](http://www.typescriptlang.org/play/#src=export%20type%20PopupEvents%20%3D%20'click'%20%7C%20'hover'%20%7C%20'focus'%0D%0Aexport%20type%20RestrictedClickEvents%20%3D%20'click'%20%7C%20'focus'%0D%0Aexport%20type%20RestrictedHoverEvents%20%3D%20'hover'%20%7C%20'focus'%0D%0Aexport%20type%20PopupEventsArray%20%3D%20RestrictedClickEvents%5B%5D%20%7C%20RestrictedHoverEvents%5B%5D%0D%0A%0D%0Atype%20OnProp%20%3D%20%20PopupEvents%20%7C%20PopupEventsArray%0D%0A%0D%0Aconst%20a%3A%20OnProp%20%3D%20%5B'click'%2C%20'click'%5D%0D%0A)

Reply:
I want to restrict using 'click' together with 'hover'. Adding the array ['click', 'click'] is not a bug, so I don't know it we want to add some other restrictions for this.